src/test/java/org/nuxeo/binary/metadata/test/TestBinaryMetadataService.java

```diff
@@ -97,6 +104,52 @@ public static void init() throws ParseException {
     }
 
     @Test
+    public void itShouldExtractSingleKeywordToStringList() throws PropertyException, IOException {
+        // Get the document with One KW attached
+    	File binary = FileUtils.getResourceFileFromContext("data/iptc_one_keyword.jpg");
+
+    	DocumentModel doc = session.createDocumentModel("/folder", "file_1000", "File");
+        doc.setPropertyValue("dc:title", "file_1000");
+        doc.setPropertyValue("file:content",  (Serializable) Blobs.createBlob(binary));
+        //doc = session.createDocument(doc);
+        binaryMetadataService.writeMetadata(doc, "IPTC-ONE-KW");
+        String[] subjects = (String[]) doc.getPropertyValue("dc:subjects");
+        assertEquals(1, subjects.length);
+
+    }
+
+    @Test
+    public void itShouldExtractKeywordListToStringList() throws PropertyException, IOException {
+        // Get the document with One KW attached
+    	File binary = FileUtils.getResourceFileFromContext("data/Budget-Example.xlsx");
+
+    	DocumentModel doc = session.createDocumentModel("/folder", "file_2000", "File");
+        doc.setPropertyValue("dc:title", "file_2000");
+        doc.setPropertyValue("file:content",  (Serializable) Blobs.createBlob(binary));
+        //doc = session.createDocument(doc);
+        binaryMetadataService.writeMetadata(doc, "EXCEL-TITLES-OF-PARTS-TO-SUBJECTS");
+        String[] subjects = (String[]) doc.getPropertyValue("dc:subjects");
+        assertEquals(4, subjects.length);
+        assertEquals("Example", subjects[0]); // ["Example","Hop","Sheet2","Chart Data"]
```

Comment:
Test the full list: `assertEquals(Arrays.asList("Example", "Hop", "Sheet2", "Chart Data"), Arrays.asList(subjects))`